src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java

```diff
@@ -434,4 +444,75 @@ else if(isFixed(loansPolicy)) {
       return Optional.empty();
     }
   }
+
+  public HttpResult<Loan> recall(Loan loan) {
+    final JsonObject recalls = representation
+        .getJsonObject("requestManagement", new JsonObject())
+        .getJsonObject("recalls", new JsonObject());
+
+    final HttpResult<DateTime> minimumDueDateResult =
+        getDueDate("minimumGuaranteedLoanPeriod", recalls,
+            loan.getLoanDate(), null);
+
+    // Use a java.time Clock, which allows unit tests to use a fixed clock
+    final DateTime systemDate = new DateTime(
+        ZonedDateTime.now(
+            ClockManager.getClockManager().getClock()).toInstant().toEpochMilli(),
+        DateTimeZone.UTC);
+
+    final HttpResult<DateTime> recallDueDateResult =
+        getDueDate("recallReturnInterval", recalls, systemDate, systemDate);
+
+    final List<ValidationError> errors = new ArrayList<>();
+
+    errors.addAll(combineValidationErrors(recallDueDateResult));
+    errors.addAll(combineValidationErrors(minimumDueDateResult));
+
+    if (errors.isEmpty()) {
```

Comment:
What happens if calculating these dates fail for a reason other than a validation error?

Reply:
@marcjohnson-kint do you see a situation where there could be a failure during the data calculation besides the ones related to `Period`. I thought I coded this up to not fail, but perhaps I missed something.

Reply:
I didn’t check, so didn’t notice any specific situations, only that it might be possible.

And could possibly be guarded against using one of the combining methods for HTTPResult

Reply:
Something like
```suggestion
      return minimumDueDateResult.combine(recallDueDateResult,
        (minimumGuaranteedDueDate, recallDueDate) -> {
          final DateTime newDueDate;
          if (minimumGuaranteedDueDate == null ||
            recallDueDate.isAfter(minimumGuaranteedDueDate)) {
            newDueDate = recallDueDate;
          } else {
            newDueDate = minimumGuaranteedDueDate;
          }

          loan.changeDueDate(newDueDate);

          return loan;
        });
```

Reply:
Or

```suggestion
    if (errors.isEmpty()) {
      return minimumDueDateResult
        .combine(recallDueDateResult,
          (minimumGuaranteedDueDate, recallDueDate) -> {
            if (minimumGuaranteedDueDate == null ||
              recallDueDate.isAfter(minimumGuaranteedDueDate)) {
               return recallDueDate;
            } else {
              return minimumGuaranteedDueDate;
            }
          })
        .map(newDueDate -> {
          loan.changeDueDate(newDueDate);

          return loan;
      });
```

Reply:
Hmm... I was under the impression nothing was actually asynchronous here and following the code path seemed to confirm that to me so I assumed `value()` would make sense here, but I might have missed something. I'll make this work with a combiner like you suggest.

Reply:
I pushed an experimental branch that might describe it better - https://github.com/folio-org/mod-circulation/commit/7d691b716940bf590fe4cb12cab24b7f689aa7bf

Reply:
I might overuse the result idea. 

It started as a way to handle failures nicer in asynchronous code, I've started using it for most things that could fail.

Reply:
You may not have missed anything, this might not happen in practice